dfttk/quasi_harmonic.py

```diff
@@ -14,44 +14,64 @@
 EV_PER_CUBIC_ANGSTROM_TO_GPA = 160.21766208  # 1 eV/Å^3  = 160.21766208 GPa
 
 
+# At the moment, the function can handle either phonon or debye. Should it handle both at the same time?
 def process_quasi_harmonic(
-    eos_parameters_df: pd.DataFrame,
-    harmonic_properties_fit: pd.DataFrame,
     volume_range: np.ndarray,
+    eos_parameters_df: pd.DataFrame,
+    harmonic_properties_fit: pd.DataFrame = None,
+    debye_properties: pd.DataFrame = None,
     thermal_electronic_properties_fit: pd.DataFrame = None,
     P: int = 0,
     eos: str = "BM4",
     plot: bool = True,
     plot_type: str = "default",
+    selected_temperatures_plot: list = None,
 ) -> pd.DataFrame:
     """Calculates the quasi-harmonic properties
 
     Args:
+        volume_range (np.ndarray): Volume range for the quasi-harmonic calculations
         eos_parameters_df (pd.DataFrame): pandas dataframe containing the EOS parameters from the eos_fit.fit_to_all function
-        harmonic_properties_fit (pd.DataFrame): pandas dataframe containing the fitted harmonic properties from the fit_harmonic function
+        harmonic_properties_fit (pd.DataFrame, optional): pandas dataframe containing the fitted harmonic properties from the fit_harmonic function
+        debye_properties (pd.DataFrame, optional): pandas dataframe containing the Debye properties. Defaults to None.
+        thermal_electronic_properties_fit (pd.DataFrame, optional): pandas dataframe containing the fitted thermal electronic properties. Defaults to None.
         P (int, optional): Pressure in GPa. Defaults to 0.
         plot (bool, optional): Defaults to True.
         plot_type (str, optional): Type of plots to include. Defaults to 'default'.
+        selected_temperatures_plot (list, optional): List of selected temperatures to plot. Defaults to None.
 
     Returns:
         pd.DataFrame: pandas dataframe containing the quasi-harmonic properties
     """
 
     # Check that all properties have the same number of atoms
     num_atoms_eos = eos_parameters_df["number_of_atoms"].values[0]
-    num_atoms_vib = harmonic_properties_fit["number_of_atoms"].values[0]
-    if thermal_electronic_properties_fit is None:
-        if num_atoms_eos != num_atoms_vib:
-            raise ValueError(
-                "The number of atoms in eos_parameters_df and harmonic_properties_fit do not match"
-            )
 
-    if thermal_electronic_properties_fit is not None:
-        num_atoms_tec = thermal_electronic_properties_fit["number_of_atoms"].values[0]
-        if num_atoms_eos != num_atoms_vib or num_atoms_eos != num_atoms_tec:
-            raise ValueError(
-                "The number of atoms in eos_parameters_df and harmonic_properties_fit do not match"
-            )
+    # Phonons only
+    if harmonic_properties_fit is not None and debye_properties is None:
```

Comment:
If the data handled separately throughout the function, we should either split into two functions. If we still want a function that is agnostic to debye or phonon data or both data, that agnostic function should then call the appropriate function debye or phonon.

Alternatively we could make the data types from debye and phonons able to be processed in the same way.

This may just be make a comment on the TODOs for now if it all currently works.

Reply:
How about I split it into:
- quasi_harmonic_phonon
- quasi_harmonic_debye

Reply:
ideally only the part that's parallel would be pulled out of the function. So a function for phonons vibration contribution a function for debye vibration contribution. Then the larger function would called one of those or both.  Also the debye model is not quasi harmonic

Reply:
Oh I guess if quasi harmonic is more general than I realized it's okay. 👍

Reply:
Ok, I will think about it.

Reply:
I will just list it as a TODO for now.